Add files via upload

alimk754 opened the pull request.

alimk754 committed:
Add files via upload

msabry1 reviewed (changes requested):
Add step by step generator

alimk754 committed:
Add files via upload

alimk754 committed:
Add files via upload

alimk754 committed:
Add files via upload

alimk754 committed:
Add files via upload

alimk754 committed:
Add files via upload

alimk754 committed:
Add files via upload

alimk754 committed:
Add files via upload

alimk754 committed:
Add files via upload

returns a generator that iterates directly through the matrix it gives every step of the u till its eliminated then gives the augmented matrix LZ=X and solve it then the augmented matrix UX=Z and solve it

alimk754 committed:
Add files via upload

contains the final result -use deconstructor in that order U,L,Forward substitution result ,final result-
